tmt/log.py

```diff
@@ -70,6 +69,65 @@ def _debug_level_from_global_envvar() -> int:
         raise tmt.utils.GeneralError(f"Invalid debug level '{raw_value}', use an integer.")
 
 
+def decide_colorization(no_color: bool, force_color: bool) -> Tuple[bool, bool]:
+    """
+    Decide whether the output and logging should be colorized.
+
+    Based on values of CLI options, environment variables and output stream
+    properties, a colorization setup is decided. The following inputs are
+    evaluated, in this order:
+
+    * if either of the ``--no-color`` CLI option, ``NO_COLOR`` or
+        ``TMT_NO_COLOR`` environment variables are set, colorization would be
+        disabled.
+    * if either of the ``--force-color`` CLI option or ``TMT_FORCE_COLOR``
+        environment variable are set, colorization would be forcefully
+        enabled.
+
+    If none of the situations above happened, colorization would be enabled for
+    output and logging based on their respective stream TTY status. Output is
+    sent to to standard output, logging then to standard error output,
+    colorization would then be the outcome of stream's :py:meth:`file.isatty`
+    method.
+
+    .. note::
+
+       Be aware that "forced enable" is stronger than "forced disable". If
+       ``--force-color`` or ``TMT_FORCE_COLOR`` are set, colors will be enabled
+       despite any disabling options or environment variables.
+
+    .. note::
+
+       All inputs with the exception of ``isatty`` result control both types of
+       output, regular output and logging, and applies to both of them. Only
+       ``isatty`` outcome is specific for each type, and may result in one
+       output type dropping colors while the other would be colorized.
+
+    :param no_color: value of the ``--no-color`` CLI option.
+    :param force_color: value of the `--force-color`` CLI option.
+    :returns: a tuple of two booleans, one for output colorization, the other
+        for logging colorization.
+    """
+
+    # Default values: assume colors & unicorns everywhere.
+    apply_colors_output, apply_colors_logging = True, True
+
+    # Disable coloring if `--no-color` was used, or `NO_COLOR` or `TMT_NO_COLOR` envvar is set.
+    if no_color or 'NO_COLOR' in os.environ or 'TMT_NO_COLOR' in os.environ:
+        apply_colors_output, apply_colors_logging = False, False
```

Comment:
Handling the `no-color` case as the first one means that it is stronger than `force-color`. Verified using:

    TMT_FORCE_COLOR=1 tmt --no-color plan ls
    <results in a no color output>

This does not match the documentation and docstring. Who's right, docs or code?

Reply:
The documentation is correct. Nice catch, fixed it and added a test for it.

Reply:
Thanks!